pkg/setup/setup.go

```diff
@@ -153,6 +162,25 @@ func (s *Setup) Configure(flags *Flags, clusterURL string, attach bool) (*config
 	return cluster, nil
 }
 
+// detectCanonicalClusterURL returns the URL with the response of the HEAD request.
+func (s *Setup) detectCanonicalClusterURL(clusterURL string, httpOpts []httpclient.Option) (string, error) {
+	httpClient := httpclient.New(clusterURL, httpOpts...)
+	req, err := httpClient.NewRequest("HEAD", "/", nil)
+	if err != nil {
+		return "", err
+	}
+
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return "", err
```

Comment:
Maybe we could make the error message more informative here, as it's the first one users should see in case of an unreachable cluster for example.